src/main/java/io/tabular/AsyncFileIO.java

```diff
@@ -412,15 +415,24 @@ public long getLength() {
       if (future.isDone()) {
         return delegate().getLength();
       } else {
+        cancelDownload();
         return source.getLength();
       }
     }
 
+    private void cancelDownload() {
+      boolean cancelled = future.cancel(true);
+      if (!cancelled) {
+        LOG.info("Failed to cancel download, this most likely means the download just completed.");
```

Comment:
I don't think this should be a warn since there's nothing really bad about this case, other than having a cache miss in the current implementation.